projects/vdk-plugins/vdk-notebook/src/vdk/plugin/notebook/cell.py

```diff
@@ -1,35 +1,42 @@
 # Copyright 2021 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
+from dataclasses import dataclass
 
 
+@dataclass
 class Cell:
     """
-    Helper class that retrieves data from Jupyter cells
-    Only the data essential for running a VDK data job is saved
-    Other data is ignored
+    A class that represents Jupyter code cell
     """
 
     def __init__(self, jupyter_cell):
         self.tags = jupyter_cell["metadata"].get("tags", {})
         self.source = "".join(jupyter_cell["source"])
 
-    def is_vdk_cell(self):
-        return "vdk" in self.tags
 
-    def is_sql_cell(self):
-        return self.source.startswith("%sql")
+class CellFuncFactory:
```

Comment:
Factory seems misnamed. When talking about factory I expect to see the design pattern - https://refactoring.guru/design-patterns/factory-method  or https://refactoring.guru/design-patterns/abstract-factory

This is more like static class with helpful utilities for cells. I'd name mit CellUtils.